src/DependencyInjection/Compiler/AddAuditEntityCompilerPass.php

```diff
@@ -48,7 +48,8 @@ public function process(ContainerBuilder $container): void
             }
 
             $definition = $container->getDefinition($id);
-            $auditedEntities[] = $this->getModelName($container, $definition->getArgument(1));
+            $modelClass = $attributes[0]['model_class'] ?? $definition->getArgument(1);
```

Comment:
I think we could add a next major comment

Reply:
Happy to, care to suggest the format for one? :)
Something like?

```suggestion
            // NEXT_MAJOR: Support only model_class and remove indexed argument support
            $modelClass = $attributes[0]['model_class'] ?? $definition->getArgument(1);
```

Reply:
Exactly, that would be nice!

Reply:
Done :+1: 

Please note that SonataAdmin also supports using `setModelClass` via `calls` in the service definition, which this code won't be compatible with, but I suppose that's part of a bigger and wider problem that this PR doesn't address.